internal/pkg/values/timespan.go

```diff
@@ -0,0 +1,186 @@
+package values
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+)
+
+var errInvalidWeekday = errors.New("error: specified weekday is invalid")
+
+type TimeSpan interface {
+	// inTimeSpan checks if time is in the timespan or not
+	isTimeInSpan(time.Time) bool
+}
+
+type timeSpans []TimeSpan
+
+// inTimeSpans checks if current time is in one of the timespans or not
+func (t *timeSpans) inTimeSpans() bool {
+	for _, timespan := range *t {
+		if !timespan.isTimeInSpan(time.Now()) {
+			continue
+		}
+		return true
+	}
+	return false
+}
+
+func (t *timeSpans) Set(value string) error {
+	spans := strings.Split(value, ",")
+	var timespans []TimeSpan
+	for _, timespanText := range spans {
+		timespanText = strings.TrimSpace(timespanText)
+
+		if isAbsoluteTimestamp(timespanText) {
+			// parse as absolute timestamp
+			timespan, err := parseAbsoluteTimeSpan(timespanText)
+			if err != nil {
+				return fmt.Errorf("failed to parse absolute timespan: %w", err)
+			}
+			timespans = append(timespans, timespan)
+			continue
+		}
+
+		// parse as relative timestamp
+		timespan, err := parseRelativeTimeSpan(timespanText)
+		if err != nil {
+			return fmt.Errorf("failed to parse relative timespan: %w", err)
+		}
+		timespans = append(timespans, timespan)
+	}
+	*t = timeSpans(timespans)
+	return nil
+}
+
+func (t *timeSpans) String() string {
+	return fmt.Sprint(*t)
+}
+
+func parseAbsoluteTimeSpan(timespan string) (absoluteTimeSpan, error) {
+	timestamps := strings.Split(timespan, " - ")
+	fromTime, err := time.Parse(time.RFC3339, timestamps[0])
+	if err != nil {
+		return absoluteTimeSpan{}, fmt.Errorf("failed to parse rfc3339 timestamp: %w", err)
+	}
+	toTime, err := time.Parse(time.RFC3339, timestamps[1])
+	if err != nil {
+		return absoluteTimeSpan{}, fmt.Errorf("failed to parse rfc3339 timestamp: %w", err)
+	}
+
+	return absoluteTimeSpan{
+		from: fromTime,
+		to:   toTime,
+	}, nil
+}
+
+func parseRelativeTimeSpan(timespanString string) (*relativeTimeSpan, error) {
+	timespan := relativeTimeSpan{}
+
+	parts := strings.Split(timespanString, " ")
+	weekdaySpan := strings.Split(parts[0], "-")
+	timeSpan := strings.Split(parts[1], "-")
+	timezone := parts[2]
+
+	var err error
+	timespan.timezone, err = time.LoadLocation(timezone)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse timezon: %w", err)
+	}
+	timespan.timeFrom, err = time.ParseInLocation("15:04", timeSpan[0], timespan.timezone)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse time: %w", err)
+	}
+	timespan.timeTo, err = time.ParseInLocation("15:04", timeSpan[1], timespan.timezone)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse time: %w", err)
+	}
+	timespan.weekdayFrom, err = getWeekday(weekdaySpan[0])
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse weekday: %w", err)
```

Comment:
We should probably include that this error stems from the weekdayFrom not parsing.